### PR TITLE
[Group 3] Adding agent and sign detection 

### PR DESCRIPTION
**Pull request by vishakhsuresh98**
This PR adds the road driving group's agent and sign detectors.

Types of objects detected:
- Agents (pedestrians and vehicles)
  - 5 classes of agents
  - PEDESTRIAN, BICYCLIST, CAR, MEDIUM_TRUCK, LARGE_TRUCK
- Signs (road signs and traffic lights):
  - 7 types of road signs and 3 types of signals
  - Signs include: NO_LEFT_TURN, NO_RIGHT_TURN, NO_U_TURN, NO_PARKING, PEDESTRIAN_CROSSING, RAILROAD_CROSSING, STOP_SIGN
  - Signals: GREEN, YELLOW, RED

### CODE REVIEW

File: GEMstack/knowledge/defaults/current.yaml
Comment (krishauser) [inline GEMstack/knowledge/defaults/current.yaml:1]: These paths should go in then gem_e4 yaml file in the vehicle directory

File: GEMstack/onboard/perception/object_detection.py
Comment (krishauser) [inline GEMstack/onboard/perception/object_detection.py:1]: How does this interact with the detection logic from group 4?

**Review by krishauser — CHANGES_REQUESTED** [review]
There are a few config paths that should be changed.

File: GEMstack/knowledge/defaults/current.yaml
Comment (vishakhsuresh98) [reply GEMstack/knowledge/defaults/current.yaml:1]: Moved them to `calibration/gem_e4.yaml`.
Should they be in `vehicle/gem_e4.yaml`?

File: GEMstack/onboard/perception/object_detection.py
Comment (vishakhsuresh98) [reply GEMstack/onboard/perception/object_detection.py:1]: `object_detection.py` does not interact with group 4's code. 

It contains most of the code common to our agent and sign detectors. More specifically, it creates a `PhysicalObject` (placed in the current frame) for each object of interest. Our agent and sign detectors use that to create an `AgentState` or `Sign` object.

Differences between the 2 versions include: 
- the object detector uses spatial clustering to identify the lidar points corresponding to a bounding box,
- our agent detector handles more classes of objects

File: GEMstack/knowledge/calibration/gem_e4.yaml
Comment (krishauser) [inline GEMstack/knowledge/calibration/gem_e4.yaml:1]: Please be more specific about which “intrinsic” this is. There are 5 cameras.

**Review by krishauser — CHANGES_REQUESTED** [review]

File: GEMstack/knowledge/calibration/gem_e4.yaml
Comment (vishakhsuresh98) [reply GEMstack/knowledge/calibration/gem_e4.yaml:1]: Moved `intrinsic` to `calibration/gem_e4_oak.yaml`.
Renamed `lidar_to_camera` as `lidar_to_oak`.

Comment (krishauser) [reply GEMstack/knowledge/calibration/gem_e4.yaml:1]: That works!

**Review by krishauser — APPROVED** [review]

**Pull request merged**